src/modules/sdk/objections/types.ts

```diff
@@ -38,12 +40,44 @@ export interface Attachment {
 }
 
 /**
- * Data structure representing an Objection
- * @interface
+ * Class representing an Objection
+ * @class
  */
-export interface Objection {
-  reason: string;
-  attachments: Array<{
+export class Objection {
+  private readonly _reason: string;
+  private readonly _attachments: Array<{
     name: string;
   }>;
+
+  constructor(reason: string, attachments: Array<{ name: string }>) {
+    this._reason = reason;
+    this._attachments = attachments;
+  }
+
+  get reason(): string {
+    return this._reason;
+  }
+
+  get attachments(): Array<{ name: string }> {
+    return this._attachments;
+  }
+
+  public getHtmlEncoded(): Objection {
+    const htmlReason: string = he.encode(this._reason);
```

Comment:
Would it be an option - and is it worth it - to HTML encode the data on object construction and store the encoded fields in the object rather than encode them every time they're requested.

Guessing this will come down to how often the 'getHtmlEncoded()' method is called (and I suspect not often) and how expensive it is.

Just though I should ask the question...

Reply:
OWASP seem to favour only encoding on output, and encoding for where it is going to be used.
If we only had the htmlEncoded version stored and needed to pass the name into js, it'd be a lot of effort to unhtml encode it, only to re URIencode it.

Reply:
I was thinking about storing both the encoded and unencoded output actually. So there'd be more fields in the object class.

Reply:
I like that idea, what if we had a setter that sets the normal field data but also encodes and populates an encoded version as another field. So 1 setter populates 2 fields. Then have 2 getters, a normal one, and getEncodedX version. That way we'll only have to run the encoder once per field.

Reply:
or, switch them around so that eg. getReason would return htmlEncoded version, and getRawReason would return the 'raw' un-encoded version (but still do the encoding in the constructor)